work_queue/src/work_queue.c

```diff
@@ -943,23 +1022,23 @@ void update_read_catalog(struct work_queue *q)
 {
 	time_t stoptime = time(0) + 5; // Short timeout for query
 
-	update_read_catalog_factory(q, stoptime);
+	if (q->fetch_factory) {
+		update_read_catalog_factory(q, stoptime);
+	}
 }
 
 void update_catalog(struct work_queue *q, struct link *foreman_uplink, int force_update )
 {
-	// Only advertise if we have a name.
-	if(!q->name) return;
-
-	// Only advertise every last_update_time seconds.
+	// Only update every last_update_time seconds.
 	if(!force_update && (time(0) - q->catalog_last_update_time) < WORK_QUEUE_UPDATE_INTERVAL)
 		return;
 
 	// If host and port are not set, pick defaults.
 	if(!q->catalog_hosts) q->catalog_hosts = xxstrdup(CATALOG_HOST);
 
 	// Update the catalog.
-	update_write_catalog(q, foreman_uplink);
+	// Only write if we have a name.
+	if (q->name) update_write_catalog(q, foreman_uplink);
 	update_read_catalog(q);
```

Comment:
Let's mirror what we do for `update_write_catalog`, e.g.:
if(q->fetch_factory) update_read_catalog(q);

Reply:
I did that inside `update_read_catalog`

```c
	if (q->fetch_factory) {
		update_read_catalog_factory(q, stoptime);
	}
```

So that we could extend this to other usages beyond reading factory conveniently.

Reply:
I see. Then let's put if(q->name)  as the first statement of `update_write_catalog`, so that both functions have the same structure.